packages/application-shell/src/components/version-tracker/version-tracker.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { Dispatch } from 'redux';
+import { useDispatch } from 'react-redux';
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import applicationShellVersion from '../../version';
+import { pushDependencyVersionCounter } from './actions';
+
+type VersionMetricOptions = {
+  applicationName?: string;
+};
+
+const createVersionMetric = ({ applicationName }: VersionMetricOptions) => [
+  {
+    metricName: 'npm_dependency_versions',
+    metricLabels: {
+      application: applicationName || 'unknown',
+      package_name: '@commercetools-frontend/application-shell',
+      package_version: applicationShellVersion,
+    },
+  },
+  {
+    metricName: 'npm_dependency_versions',
+    metricLabels: {
+      application: applicationName || 'unknown',
+      package_name: 'react',
+      package_version: React.version,
+    },
+  },
+];
+
+const VersionTracker = () => {
+  const applicationName = useApplicationContext(
+    context => context.environment.applicationName
+  );
+  const dispatch = useDispatch<
+    Dispatch<ReturnType<typeof pushDependencyVersionCounter>>
+  >();
+  React.useEffect(() => {
+    const result = (dispatch(
+      pushDependencyVersionCounter({
+        payload: createVersionMetric({
+          applicationName,
+        }),
+      })
+    ) as unknown) as Promise<unknown>;
```

Comment:
FYI: I bumped into this issue and therefore noticed that the SDK types still need to be adjusted.
The main issue is that the SDK creates “normal” plain actions but the middleware returns a Promise with the HTTP response. When using `dispatch`, it doesn’t know anything about the return type, unless otherwise specified, which means that we need to “patch it up”. For example we can expose a `useAsyncDispatch` which just dispatches the action normally but provides the "correct” types.
I’ll do that in a separate PR.